src/manifest.rs

```diff
@@ -183,6 +164,75 @@ impl Manifest {
         data.parse()
     }
 
+    /// Get the specified table from the manifest.
+    pub fn get_table<'a>(
+        &'a mut self,
+        table_path: &[String],
+    ) -> Result<&'a mut BTreeMap<String, toml::Value>, ManifestError> {
+
+        /// Descend into a manifest until the required table is found.
+        fn descend<'a>(
+            input: &'a mut BTreeMap<String, toml::Value>,
+            mut path: VecDeque<&String>,
+        ) -> Result<&'a mut BTreeMap<String, toml::Value>, ManifestError> {
+            if let Some(segment) = path.pop_front() {
+                let value = match *input
+                    .entry(segment.to_owned())
+                    .or_insert_with(|| toml::Value::Table(BTreeMap::new())) {
```

Comment:
This is hard to read. Can you give the thing you are matching over its own binding and keep the match head single-line?